hermes-common/src/test/groovy/pl/allegro/tech/hermes/infrastructure/zookeeper/ZookeeperWorkloadConstraintsCacheTest.groovy

```diff
@@ -0,0 +1,166 @@
+
+        when:
+        deleteData('/hermes/consumers-workload-constraints/group.topic')
+        ensureCacheWasUpdated(1)
+        def updatedConstraints = pathChildrenCache.getConsumersWorkloadConstraints()
+
+        then:
+        updatedConstraints.topicConstraints == emptyMap()
+        constraints.subscriptionConstraints.get(SubscriptionName.fromString('group.topic$sub')).consumersNumber == 3
+    }
+
+    def "should update cache on change node"() {
+        given:
+        setupNode('/hermes/consumers-workload-constraints/group.topic', new Constraints(1))
+        setupNode('/hermes/consumers-workload-constraints/group.topic$sub', new Constraints(3))
+        ensureCacheWasUpdated(2)
+
+        when:
+        def constraints = pathChildrenCache.getConsumersWorkloadConstraints()
+
+        then:
+        constraints.topicConstraints.get(TopicName.fromQualifiedName('group.topic')).consumersNumber == 1
+        constraints.subscriptionConstraints.get(SubscriptionName.fromString('group.topic$sub')).consumersNumber == 3
+
+        when:
+        updateNode('/hermes/consumers-workload-constraints/group.topic', new Constraints(2))
+        ensureCacheWasUpdated(2)
+
+        def updatedConstraints = pathChildrenCache.getConsumersWorkloadConstraints()
+
+        then:
+        updatedConstraints.topicConstraints.get(TopicName.fromQualifiedName('group.topic')).consumersNumber == 2
+        updatedConstraints.subscriptionConstraints.get(SubscriptionName.fromString('group.topic$sub')).consumersNumber == 3
+    }
+
+    def "should log error if cannot read data from node"() {
+        given:
+        setupNode('/hermes/consumers-workload-constraints/group.topic', new Constraints(1))
+        setupNode('/hermes/consumers-workload-constraints/group.topic$sub', new Constraints(3))
+        ensureCacheWasUpdated(2)
+
+        when:
+        def constraints = pathChildrenCache.getConsumersWorkloadConstraints()
+
+        then:
+        constraints.topicConstraints.get(TopicName.fromQualifiedName('group.topic')).consumersNumber == 1
+        constraints.subscriptionConstraints.get(SubscriptionName.fromString('group.topic$sub')).consumersNumber == 3
+
+        when:
+        updateNode('/hermes/consumers-workload-constraints/group.topic', 'random data')
+        ensureCacheWasUpdated(2)
+
+        def updatedConstraints = pathChildrenCache.getConsumersWorkloadConstraints()
+
+        then: 'data remained intact'
+        updatedConstraints.topicConstraints.get(TopicName.fromQualifiedName('group.topic')).consumersNumber == 1
+        updatedConstraints.subscriptionConstraints.get(SubscriptionName.fromString('group.topic$sub')).consumersNumber == 3
+
+        and:
+        listAppender.list.get(0).formattedMessage == 'Cannot read data from node: /hermes/consumers-workload-constraints/group.topic'
+        listAppender.list.get(0).throwableProxy.className == 'com.fasterxml.jackson.databind.exc.MismatchedInputException'
```

Comment:
I think it's better to check if the actual exception was thrown, instead of checking logs

Reply:
The exception is caught, so I can't check if it was thrown.